rpc/transport/ws/server.go

```diff
@@ -77,10 +81,24 @@ func (wsc *serverWebSocketTransport) Close() {
 }
 
 func (wsc *serverWebSocketTransport) Url() string {
-	return webscocketServerAddress + wsc.port
+	return websocketServerAddress + wsc.port + apiVersionPath
 }
 
 func (wsc *serverWebSocketTransport) request(w http.ResponseWriter, r *http.Request) {
+	// Pull api version from the url and determine if the version is supported
```

Comment:
Have we given any thought to how we plan on versioning the API? For example, does it make sense to use semantic versioning or not?

Reply:
To give us the highest likelihood for compatibility between go-nitro nodes and clients, I am proposing that we just use major api versions in the jsonrpc server url. If there is a breaking change, we can bump the api version (and maybe servers support at least one older version of the api?). If there is a non-breaking change (i.e. a minor revision bump) I don't think we would want to change the jsonrpc server endpoint because clients would have to explicitly change their requests. Also it seems like a single go-nitro might have to simultaneously support too many versions of the api (meaning lots of duplicated code) if we are differentiating between the minor revisions through the url. Maybe there is still a way to check the go-nitro node's overall version (through the commit hash or build tag), which could help differentiate between jsonrpc api minor versions without breaking compatibility.

That was my thinking, but happy to change if you feel like it should be built differently.

Reply:
Sounds like good thinking to me.